tests/lib/nested.sh

```diff
@@ -68,6 +68,35 @@ nested_wait_for_snap_command() {
     done
 }
 
+nested_check_unit_active() {
+    local nested_unit="${1:-$NESTED_VM}"
+    local retry=${2:-5}
+    local wait=${3:-1}
+
+    while [ "$retry" -ge 0 ]; do
+        retry=$(( retry - 1 ))
+
+        if ! systemctl is-active "$nested_unit"; then
+            echo "Unit $nested_unit is not active. Aborting!"
+            return 1
+        fi
+        sleep "$wait"
+    done
+}
+
+nested_check_boot_errors() {
+    # make sure the service started and it is running
+    if nested_is_core_20_system && ! nested_check_unit_active "$NESTED_VM" 15 1; then
+        if nested_status_vm | grep -qE "cpu_asidx_from_attrs: Assertion.*failed"; then
```

Comment:
could you add a comment explaining a little bit about the intention of this? it's my understanding that restarting the VM here is necessary because in this state the hypervisor is essentially frozen/broken and we need to restart it to kick it in the hopes that it recovers, is that correct? Also could the hypervisor run into the same issue again even after being restarted? Or has that ever been noticed to happen where we restart it and it gets stuck again?

Reply:
Hi, I added the commnent about the error, I also added a retry in case it happens twice.
Thanks for the review